irc_callback.go

```diff
@@ -232,33 +232,88 @@ func (irc *Connection) setupCallbacks() {
 	// work. It has to be set somewhere first in case the nick is already
 	// taken or unavailable from the beginning.
 	irc.AddCallback("437", func(e *Event) {
-		// If irc.nickcurrent hasn't been set yet, set to irc.nick
-		if irc.nickcurrent == "" {
-			irc.nickcurrent = irc.nick
+		irc.Lock()
+		defer irc.Unlock()
+		if !irc.fully_connected {
+			if irc.nickcurrent == "" {
+				irc.nickcurrent = irc.nick
+			}
+			if len(irc.nickcurrent) > 8 {
+				irc.nickcurrent = "_" + irc.nickcurrent
+			} else {
+				irc.nickcurrent = irc.nickcurrent + "_"
+			}
```

Comment:
I would create a function for this, the exact same code is used four times.

Reply:
I will try to do that in my next code update for this fork.